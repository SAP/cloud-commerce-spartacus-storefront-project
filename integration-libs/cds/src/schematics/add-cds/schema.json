{
  "$schema": "http://json-schema.org/schema",
  "id": "CDSSchematics",
  "title": "CDS Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "tenant": {
      "type": "string",
      "description": "The tenant name.",
      "default": "my-tenant",
      "x-prompt": "What is your tenant name?"
    },
    "baseUrl": {
      "type": "string",
      "description": "The base URL.",
      "default": "https://my-tenant.api.us.context.cloud.sap",
      "x-prompt": "What is your base URL?"
    },
    "profileTagLoadUrl": {
      "type": "string",
      "description": "Where to load ProfileTag from. (e.g. https://tag.static.us.context.cloud.sap/js/profile-tag.js)",
      "x-prompt": "Where to load ProfileTag from? (optional)"
    },
    "profileTagConfigUrl": {
      "type": "string",
      "description": " Configuration URL for ProfileTag. (e.g. https://tag.static.stage.context.cloud.sap/config/my-config123)",
      "x-prompt": "What is the ProfileTag configuration URL? (optional)"
    }
  },
  "required": ["tenant", "baseUrl"]
}
