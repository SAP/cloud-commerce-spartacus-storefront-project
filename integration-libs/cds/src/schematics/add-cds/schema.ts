import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  tenant: string;
  baseUrl: string;
  strategyId: string;
  profileTagLoadUrl?: string;
  profileTagConfigUrl?: string;
}
