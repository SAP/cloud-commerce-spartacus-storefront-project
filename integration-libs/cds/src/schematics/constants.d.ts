export declare const CDS_FOLDER_NAME = "cds";
export declare const CDS_MODULE = "CdsModule";
export declare const CLI_CDS_FEATURE = "cds";
