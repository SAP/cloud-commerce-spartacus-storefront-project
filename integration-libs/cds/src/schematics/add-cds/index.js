"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addCdsFeature = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const schematics_2 = require("@spartacus/schematics");
const package_json_1 = require("../../../package.json");
const constants_1 = require("../constants");
function addCdsFeature(options) {
    return (tree, _context) => {
        const packageJson = schematics_2.readPackageJson(tree);
        schematics_2.validateSpartacusInstallation(packageJson);
        validateCdsOptions(options);
        return schematics_1.chain([
            addCds(options),
            addCdsPackageJsonDependencies(packageJson),
            schematics_2.installPackageJsonDependencies(),
        ]);
    };
}
exports.addCdsFeature = addCdsFeature;
function validateCdsOptions({ tenant, baseUrl }) {
    if (!tenant) {
        throw new schematics_1.SchematicsException(`Please specify tenant name.`);
    }
    if (!baseUrl) {
        throw new schematics_1.SchematicsException(`Please specify the base URL.`);
    }
}
function addCdsPackageJsonDependencies(packageJson) {
    const dependencies = schematics_2.createDependencies(package_json_1.peerDependencies);
    return schematics_2.addPackageJsonDependencies(dependencies, packageJson);
}
function addCds(options) {
    const customConfig = [
        {
            import: [
                {
                    moduleSpecifier: schematics_2.SPARTACUS_CDS,
                    namedImports: [schematics_2.CDS_CONFIG],
                },
            ],
            content: `<${schematics_2.CDS_CONFIG}>{
      cds: {
        tenant: '${options.tenant}',
        baseUrl: '${options.baseUrl}',
        endpoints: {
          strategyProducts: '/strategy/\${tenant}/strategies/\${strategyId}/products',
        },
        merchandising: {
          defaultCarouselViewportThreshold: 80,
        },
      },
    }`,
        },
    ];
    if (options.profileTagLoadUrl && options.profileTagConfigUrl) {
        customConfig.push({
            import: [
                {
                    moduleSpecifier: schematics_2.SPARTACUS_CDS,
                    namedImports: [schematics_2.CDS_CONFIG],
                },
            ],
            content: `<${schematics_2.CDS_CONFIG}>{
          cds: {
            profileTag: {
              javascriptUrl:
                '${options.profileTagLoadUrl}',
              configUrl:
                '${options.profileTagConfigUrl}',
              allowInsecureCookies: true,
            },
          },
        }`,
        });
    }
    return schematics_2.addLibraryFeature(Object.assign(Object.assign({}, options), { lazy: false }), {
        folderName: constants_1.CDS_FOLDER_NAME,
        name: constants_1.CLI_CDS_FEATURE,
        featureModule: {
            importPath: schematics_2.SPARTACUS_CDS,
            name: constants_1.CDS_MODULE,
            content: `${constants_1.CDS_MODULE}.forRoot()`,
        },
        customConfig,
    });
}
//# sourceMappingURL=index.js.map