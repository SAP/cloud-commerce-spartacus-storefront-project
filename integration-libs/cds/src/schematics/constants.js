"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CLI_CDS_FEATURE = exports.CDS_MODULE = exports.CDS_FOLDER_NAME = void 0;
exports.CDS_FOLDER_NAME = 'cds';
exports.CDS_MODULE = 'CdsModule';
exports.CLI_CDS_FEATURE = 'cds';
//# sourceMappingURL=constants.js.map