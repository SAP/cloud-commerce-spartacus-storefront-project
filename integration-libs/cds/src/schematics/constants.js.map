{"version":3,"file":"constants.js","sourceRoot":"","sources":["constants.ts"],"names":[],"mappings":";;;AAAa,QAAA,eAAe,GAAG,KAAK,CAAC;AAExB,QAAA,UAAU,GAAG,WAAW,CAAC;AAEzB,QAAA,eAAe,GAAG,KAAK,CAAC"}