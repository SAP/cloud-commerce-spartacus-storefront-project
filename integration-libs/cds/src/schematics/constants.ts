export const CDS_FOLDER_NAME = 'cds';

export const CDS_MODULE = 'CdsModule';

export const CLI_CDS_FEATURE = 'cds';
