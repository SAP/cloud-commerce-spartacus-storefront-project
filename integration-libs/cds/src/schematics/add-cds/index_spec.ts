/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import { SpartacusOptions } from '@spartacus/schematics';
import * as path from 'path';
import { Schema as SpartacusCdsOptions } from './schema';

const collectionPath = path.join(__dirname, '../collection.json');
const spartacusCdsFeatureModulePath =
  'src/app/spartacus/features/cds/cds-feature.module.ts';

describe('Spartacus CDS schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const defaultOptions: SpartacusCdsOptions = {
    project: 'schematics-test',
    lazy: true,
    tenant: 'my-tenant',
    baseUrl: 'my-base-url.com',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    configuration: 'b2c',
    lazy: true,
    features: [],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      '@spartacus/schematics',
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@spartacus/schematics',
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('CDS feature', () => {
    describe('general setup without ProfileTag', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultOptions, appTree)
          .toPromise();
      });

      it('should import feature module in SpartacusFeaturesModule', () => {
        const spartacusFeaturesModulePath = appTree.readContent(
          'src/app/spartacus/spartacus-features.module.ts'
        );
        expect(spartacusFeaturesModulePath).toMatchSnapshot();
      });

      it('should generate CDS feature module without ProfileTag config', () => {
        const cdsFeatureModule = appTree.readContent(
          spartacusCdsFeatureModulePath
        );
        expect(cdsFeatureModule).toMatchSnapshot();
      });
    });

    describe('general setup with ProfileTag', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            {
              ...defaultOptions,
              profileTagConfigUrl: 'profile-tag-config-url.com',
              profileTagLoadUrl: 'profile-tag-load-url.com',
            },
            appTree
          )
          .toPromise();
      });

      it('should import feature module in SpartacusFeaturesModule', () => {
        const spartacusFeaturesModulePath = appTree.readContent(
          'src/app/spartacus/spartacus-features.module.ts'
        );
        expect(spartacusFeaturesModulePath).toMatchSnapshot();
      });
      it('should generate CDS feature module with ProfileTag config', () => {
        const cdsFeatureModule = appTree.readContent(
          spartacusCdsFeatureModulePath
        );
        expect(cdsFeatureModule).toMatchSnapshot();
      });
    });
  });
});
