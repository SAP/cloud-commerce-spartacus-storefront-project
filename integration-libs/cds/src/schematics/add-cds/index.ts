import {
  chain,
  Rule,
  SchematicContext,
  SchematicsException,
  Tree,
} from '@angular-devkit/schematics';
import {
  addLibraryFeature,
  addPackageJsonDependencies,
  CDS_CONFIG,
  createDependencies,
  CustomConfig,
  installPackageJsonDependencies,
  readPackageJson,
  SPARTACUS_CDS,
  validateSpartacusInstallation,
} from '@spartacus/schematics';
import { peerDependencies } from '../../../package.json';
import { CDS_FOLDER_NAME, CDS_MODULE, CLI_CDS_FEATURE } from '../constants';
import { Schema as SpartacusCdsOptions } from './schema';

export function addCdsFeature(options: SpartacusCdsOptions): Rule {
  return (tree: Tree, _context: SchematicContext) => {
    const packageJson = readPackageJson(tree);
    validateSpartacusInstallation(packageJson);
    validateCdsOptions(options);

    return chain([
      addCds(options),

      addCdsPackageJsonDependencies(packageJson),
      installPackageJsonDependencies(),
    ]);
  };
}

function validateCdsOptions({ tenant, baseUrl }: SpartacusCdsOptions): void {
  if (!tenant) {
    throw new SchematicsException(`Please specify tenant name.`);
  }
  if (!baseUrl) {
    throw new SchematicsException(`Please specify the base URL.`);
  }
}

function addCdsPackageJsonDependencies(packageJson: any): Rule {
  const dependencies = createDependencies(peerDependencies);
  return addPackageJsonDependencies(dependencies, packageJson);
}

function addCds(options: SpartacusCdsOptions): Rule {
  const customConfig: CustomConfig[] = [
    {
      import: [
        {
          moduleSpecifier: SPARTACUS_CDS,
          namedImports: [CDS_CONFIG],
        },
      ],
      content: `<${CDS_CONFIG}>{
      cds: {
        tenant: '${options.tenant}',
        baseUrl: '${options.baseUrl}',
        endpoints: {
          strategyProducts: '/strategy/\${tenant}/strategies/\${strategyId}/products',
        },
        merchandising: {
          defaultCarouselViewportThreshold: 80,
        },
      },
    }`,
    },
  ];
  if (options.profileTagLoadUrl && options.profileTagConfigUrl) {
    customConfig.push({
      import: [
        {
          moduleSpecifier: SPARTACUS_CDS,
          namedImports: [CDS_CONFIG],
        },
      ],
      content: `<${CDS_CONFIG}>{
          cds: {
            profileTag: {
              javascriptUrl:
                '${options.profileTagLoadUrl}',
              configUrl:
                '${options.profileTagConfigUrl}',
              allowInsecureCookies: true,
            },
          },
        }`,
    });
  }

  return addLibraryFeature(
    { ...options, lazy: false },
    {
      folderName: CDS_FOLDER_NAME,
      name: CLI_CDS_FEATURE,
      featureModule: {
        importPath: SPARTACUS_CDS,
        name: CDS_MODULE,
        content: `${CDS_MODULE}.forRoot()`,
      },
      customConfig,
    }
  );
}
