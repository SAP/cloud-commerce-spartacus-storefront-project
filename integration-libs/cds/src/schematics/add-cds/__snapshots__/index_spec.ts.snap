// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus CDS schematics: ng-add CDS feature general setup with ProfileTag should generate CDS feature module with ProfileTag config 1`] = `
"import { NgModule } from '@angular/core';
import { CdsConfig, CdsModule } from \\"@spartacus/cds\\";
import { provideConfig } from \\"@spartacus/core\\";

@NgModule({
  declarations: [],
  imports: [
    CdsModule.forRoot()
  ],
  providers: [provideConfig(<CdsConfig>{
    cds: {
      tenant: 'my-tenant',
      baseUrl: 'my-base-url.com',
      endpoints: {
        strategyProducts: '/strategy/\${tenant}/strategies/\${strategyId}/products',
      },
      merchandising: {
        defaultCarouselViewportThreshold: 80,
      },
    },
  }),
  provideConfig(<CdsConfig>{
    cds: {
      profileTag: {
        javascriptUrl:
          'profile-tag-load-url.com',
        configUrl:
          'profile-tag-config-url.com',
        allowInsecureCookies: true,
      },
    },
  })
  ]
})
export class CdsFeatureModule { }
"
`;

exports[`Spartacus CDS schematics: ng-add CDS feature general setup with ProfileTag should import feature module in SpartacusFeaturesModule 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CartModule, CartOccModule, CheckoutModule, CheckoutOccModule, CostCenterOccModule, ProductModule, ProductOccModule, UserOccTransitionalModule, UserTransitionalModule } from \\"@spartacus/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CartComponentModule, CartPageEventModule, CategoryNavigationModule, CheckoutComponentModule, CheckoutLoginModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, OrderCancellationModule, OrderConfirmationModule, OrderDetailsModule, OrderHistoryModule, OrderReturnModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ReplenishmentOrderConfirmationModule, ReplenishmentOrderDetailsModule, ReplenishmentOrderHistoryModule, ReturnRequestDetailModule, ReturnRequestListModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule, WishListModule } from \\"@spartacus/storefront\\";
import { CdsFeatureModule } from './features/cds/cds-feature.module';

@NgModule({
  declarations: [],
  imports: [
    // Auth Core
    AuthModule.forRoot(),
    LogoutModule,
    LoginRouteModule,
    // Basic Cms Components
    HamburgerMenuModule,
    SiteContextSelectorModule,
    LinkModule,
    BannerModule,
    CmsParagraphModule,
    TabParagraphContainerModule,
    BannerCarouselModule,
    CategoryNavigationModule,
    NavigationModule,
    FooterNavigationModule,
    BreadcrumbModule,
    // User Core,
    UserTransitionalModule,
    UserOccTransitionalModule,
    // User UI,
    AddressBookModule,
    PaymentMethodsModule,
    NotificationPreferenceModule,
    MyInterestsModule,
    StockNotificationModule,
    ConsentManagementModule,
    MyCouponsModule,
    // Anonymous Consents Core,
    AnonymousConsentsModule.forRoot(),
    // Anonymous Consents UI,
    AnonymousConsentsDialogModule,
    AnonymousConsentManagementBannerModule,
    // Product Core,
    ProductModule.forRoot(),
    ProductOccModule,
    // Product UI,
    ProductDetailsPageModule,
    ProductListingPageModule,
    ProductListModule,
    SearchBoxModule,
    ProductFacetNavigationModule,
    ProductTabsModule,
    ProductCarouselModule,
    ProductReferencesModule,
    ProductImagesModule,
    ProductSummaryModule,
    ProductIntroModule,
    // Cart Core,
    CartModule.forRoot(),
    CartOccModule,
    // Cart UI,
    CartComponentModule,
    WishListModule,
    // Checkout Core,
    CheckoutModule.forRoot(),
    CheckoutOccModule,
    CostCenterOccModule,
    // Checkout UI,
    CheckoutLoginModule,
    CheckoutComponentModule,
    OrderConfirmationModule,
    // Order,
    OrderHistoryModule,
    OrderDetailsModule,
    OrderCancellationModule,
    OrderReturnModule,
    ReturnRequestListModule,
    ReturnRequestDetailModule,
    ReplenishmentOrderHistoryModule,
    ReplenishmentOrderDetailsModule,
    ReplenishmentOrderConfirmationModule,
    // Page Events,
    NavigationEventModule,
    HomePageEventModule,
    CartPageEventModule,
    ProductPageEventModule,
    CdsFeatureModule,
  ]
})
export class SpartacusFeaturesModule { }
"
`;

exports[`Spartacus CDS schematics: ng-add CDS feature general setup without ProfileTag should generate CDS feature module without ProfileTag config 1`] = `
"import { NgModule } from '@angular/core';
import { CdsConfig, CdsModule } from \\"@spartacus/cds\\";
import { provideConfig } from \\"@spartacus/core\\";

@NgModule({
  declarations: [],
  imports: [
    CdsModule.forRoot()
  ],
  providers: [provideConfig(<CdsConfig>{
    cds: {
      tenant: 'my-tenant',
      baseUrl: 'my-base-url.com',
      endpoints: {
        strategyProducts: '/strategy/\${tenant}/strategies/\${strategyId}/products',
      },
      merchandising: {
        defaultCarouselViewportThreshold: 80,
      },
    },
  })]
})
export class CdsFeatureModule { }
"
`;

exports[`Spartacus CDS schematics: ng-add CDS feature general setup without ProfileTag should import feature module in SpartacusFeaturesModule 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CartModule, CartOccModule, CheckoutModule, CheckoutOccModule, CostCenterOccModule, ProductModule, ProductOccModule, UserOccTransitionalModule, UserTransitionalModule } from \\"@spartacus/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CartComponentModule, CartPageEventModule, CategoryNavigationModule, CheckoutComponentModule, CheckoutLoginModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, OrderCancellationModule, OrderConfirmationModule, OrderDetailsModule, OrderHistoryModule, OrderReturnModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ReplenishmentOrderConfirmationModule, ReplenishmentOrderDetailsModule, ReplenishmentOrderHistoryModule, ReturnRequestDetailModule, ReturnRequestListModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule, WishListModule } from \\"@spartacus/storefront\\";
import { CdsFeatureModule } from './features/cds/cds-feature.module';

@NgModule({
  declarations: [],
  imports: [
    // Auth Core
    AuthModule.forRoot(),
    LogoutModule,
    LoginRouteModule,
    // Basic Cms Components
    HamburgerMenuModule,
    SiteContextSelectorModule,
    LinkModule,
    BannerModule,
    CmsParagraphModule,
    TabParagraphContainerModule,
    BannerCarouselModule,
    CategoryNavigationModule,
    NavigationModule,
    FooterNavigationModule,
    BreadcrumbModule,
    // User Core,
    UserTransitionalModule,
    UserOccTransitionalModule,
    // User UI,
    AddressBookModule,
    PaymentMethodsModule,
    NotificationPreferenceModule,
    MyInterestsModule,
    StockNotificationModule,
    ConsentManagementModule,
    MyCouponsModule,
    // Anonymous Consents Core,
    AnonymousConsentsModule.forRoot(),
    // Anonymous Consents UI,
    AnonymousConsentsDialogModule,
    AnonymousConsentManagementBannerModule,
    // Product Core,
    ProductModule.forRoot(),
    ProductOccModule,
    // Product UI,
    ProductDetailsPageModule,
    ProductListingPageModule,
    ProductListModule,
    SearchBoxModule,
    ProductFacetNavigationModule,
    ProductTabsModule,
    ProductCarouselModule,
    ProductReferencesModule,
    ProductImagesModule,
    ProductSummaryModule,
    ProductIntroModule,
    // Cart Core,
    CartModule.forRoot(),
    CartOccModule,
    // Cart UI,
    CartComponentModule,
    WishListModule,
    // Checkout Core,
    CheckoutModule.forRoot(),
    CheckoutOccModule,
    CostCenterOccModule,
    // Checkout UI,
    CheckoutLoginModule,
    CheckoutComponentModule,
    OrderConfirmationModule,
    // Order,
    OrderHistoryModule,
    OrderDetailsModule,
    OrderCancellationModule,
    OrderReturnModule,
    ReturnRequestListModule,
    ReturnRequestDetailModule,
    ReplenishmentOrderHistoryModule,
    ReplenishmentOrderDetailsModule,
    ReplenishmentOrderConfirmationModule,
    // Page Events,
    NavigationEventModule,
    HomePageEventModule,
    CartPageEventModule,
    ProductPageEventModule,
    CdsFeatureModule,
  ]
})
export class SpartacusFeaturesModule { }
"
`;
